frontend/src/views/FuelCodes/AddFuelCode/_schema.jsx

```diff
@@ -452,36 +446,33 @@ export const fuelCodeColDefs = (optionsData, errors, isCreate, canEdit) => [
     headerName: i18n.t(
       'fuelCode:fuelCodeColLabels.fuelProductionFacilityProvinceState'
     ),
-    cellEditor: AutocompleteCellEditor,
+    cellEditor: AsyncSuggestionEditor,
     suppressKeyboardEvent,
     cellDataType: 'text',
     cellRenderer: createCellRenderer('fuelProductionFacilityProvinceState'),
-    cellEditorParams: {
-      onDynamicUpdate: (val, params) => params.api.stopEditing(),
-      noLabel: true,
-      options: [
-        ...new Map(
-          optionsData.fpLocations.map((location) => [
-            location.fuelProductionFacilityProvinceState,
-            location.fuelProductionFacilityProvinceState
-          ])
-        ).values()
-      ],
-      multiple: false,
-      disableCloseOnSelect: false,
-      freeSolo: true,
-      openOnFocus: true
-    },
+    cellEditorParams: (params) => ({
+      queryKey: 'fuel-production-province-search',
+      queryFn: async ({ queryKey, client }) => {
+        let path = apiRoutes.fuelCodeSearch
+        path += 'fpProvince=' + queryKey[1]
+        const response = await client.get(path)
+        return response.data
+      },
+      title: 'fuelProductionFacilityProvinceState',
+      enabled: true
+    }),
     minWidth: 325,
     valueSetter: (params) => {
-      params.data.fuelProductionFacilityProvinceState = params.newValue
+      if (!params.newValue) return false // Handle empty values safely
```

Comment:
Do we still need to set the empty value on params.data? Is it leaving the old value around?